src/main/java/com/klodnicki/bike/rest/API/Bike/rest/controller/UserController.java

```diff
@@ -1,12 +1,47 @@
 package com.klodnicki.bike.rest.API.Bike.rest.controller;
 
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.klodnicki.bike.rest.API.Bike.exception.NotFoundInDatabaseException;
+import com.klodnicki.bike.rest.API.Bike.model.entity.User;
+import com.klodnicki.bike.rest.API.Bike.service.UserService;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
-@RequestMapping("/user")
+@RequestMapping("/api/user")
 public class UserController {
 
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/add")
+    public User addUser(@RequestBody User user) {
+        return userService.addUser(user);
+    }
+
+    @GetMapping("/{id}") //dostawałem 400 bad request bo była zła adnotacja, była @RequestParam a powinna być @PathVariable
+    // i wtedy daję: localhost:8888/api/user/2
+    public User findUserById (@PathVariable("id") Long id) throws NotFoundInDatabaseException {
+        return userService.findUserById(id);
+    }
+
+    @GetMapping
+    public Iterable<User> findAllUsers() {
+        return userService.findAllUsers();
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteUserById (@PathVariable("id") Long id) throws NotFoundInDatabaseException {
```

Comment:
It would be much better to return something here, so we can see in postman what has been done exactly. If the return type is void and only status 200 OK is send I assume, then we can expect that the object was just deleted. But are there any more consequences to that? Why void exactly in your opinion?

Reply:
here is similar, the default crud method does not return anything, and also me - I got used to get 200 OK response and no response body

![image](https://github.com/BartoliniAndBorderCollies/Rent_bike_REST_API_Official/assets/126821059/4c2a378d-5963-42e1-b82d-2149a11c223b)

Reply:
Okay then, that's fine for me :)